presto-hive/src/main/java/com/facebook/presto/hive/s3select/S3SelectLineRecordReader.java

```diff
@@ -130,6 +131,8 @@
 
     protected abstract OutputSerialization buildOutputSerialization();
 
+    protected abstract boolean shouldEnableScanRange();
```

Comment:
It seems like this can all be driven by `HiveFileSplit#isSplittable()` (as determined by `InternalHiveSplitFactory#createInternalHiveSplit`). After that point we've statically already determined either to generate sub-file splits or a single split for the whole file and it seems like we can leverage that determination instead of re-checking it here.

Reply:
Thanks, updated! Please check when you find the time 👍